Userland/Libraries/LibGL/Vertex.cpp

```diff
@@ -289,7 +311,12 @@ void GLContext::gl_vertex_pointer(GLint size, GLenum type, GLsizei stride, void
     RETURN_WITH_ERROR_IF(!(type == GL_SHORT || type == GL_INT || type == GL_FLOAT || type == GL_DOUBLE), GL_INVALID_ENUM);
     RETURN_WITH_ERROR_IF(stride < 0, GL_INVALID_VALUE);
 
-    m_client_vertex_pointer = { .size = size, .type = type, .normalize = false, .stride = stride, .pointer = pointer };
+    if (m_array_buffer) {
+        size_t data_offset = (size_t)pointer;
+        m_client_vertex_pointer = { .size = size, .type = type, .normalize = false, .stride = stride, .pointer = m_array_buffer->offset_data(data_offset) };
+    } else {
+        m_client_vertex_pointer = { .size = size, .type = type, .normalize = false, .stride = stride, .pointer = pointer };
+    }
```

Comment:
Code style: if the only thing you're varying is the `.pointer` member in the struct, maybe use a variable instead of duplicating all other struct members with the same value? My opinion, I'm also fine with keeping it this way.